src/phoenix/trace/llama_index/callback.py

```diff
@@ -265,17 +292,68 @@ def get_spans(self) -> Iterator[Span]:
         return self._tracer.get_spans()
 
 
-def _add_to_tracer(
-    event_id_to_event_data: Dict[CBEventID, CBEventData],
-    trace_map: Dict[CBEventID, List[CBEventID]],
+def _process_exceptions(
+    event_id_to_event_data: EventMap, trace_map: TraceMap
+) -> Tuple[EventMap, TraceMap]:
+    """
+    Processes and removes exception events from the trace map. Returns copies of the inputs in which
+    the exception data has been added to the appropriate event payload and the exception events have
+    been removed.
+
+    As context, exception events are siblings and not children of the event that caused the
+    exception. There is a bug within LlamaIndex causing duplicate exception events to be emitted for
+    a single exception. This function uses a right-to-left BFS to collect exception events and then
+    associate the first exception in a sequence with the event that caused it.
+    """
+    parent_event_id = "root"
+    ids_queue = deque(
+        [(parent_event_id, event_id) for event_id in reversed(trace_map[parent_event_id])]
+    )
+    span_exception = None
```

Comment:
Is it correct to assume that there can only be _one_ exception?

Reply:
There's a bug in LlamaIndex right now that looks like it's causing the same exception event to be registered multiple time. This is to get around that issue.

Reply:
Probably can just let the bug go through, because the bug is only temporary and will go away, but the structural assumption here of having just a single exception seems too strong.

Reply:
addressed in [b8dc1e4](https://github.com/Arize-ai/phoenix/pull/1449/commits/b8dc1e423ee7b8d596d949963afc14d6f9fa2a6a)